### PR TITLE
Enhance GROMACS workflow

### PR DESCRIPTION
**Pull request by nebilbenmabrouk**

### CODE REVIEW

**Review by aminelouati — APPROVED** [review]

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 1]
```diff
@@ -6,15 +6,24 @@
...
     <variable name="GROMACS_PORT" value="9999" />
     <variable name="GROMACS_MPI_THREADS" value="2" />
-    <variable name="GROMACS_OPEN_MPI_THREADS" value="2" />
+    <variable name="GROMACS_OPENMP_THREADS" value="2" />
   </variables>
   <description>
-    <![CDATA[ Example of Remote Visualization using Xvnc. It uses the following parameters:
-DISPLAY: display number to use
-VISU_COMMAND: vnc command to start ]]>
+    <![CDATA[ Example of a HPC workflow, which performs a molecular behavior simulation and visualize it, using the platforms GROMACS, VMD and Xvnc.
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:12]: Example of an HPC workflow, which performs a molecular behavior simulation and visualization using the platforms GROMACS, VMD and Xvnc.

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 2]
```diff
@@ -6,15 +6,24 @@
...
+- GROMACS: a platform which performs molecular dynamics (https://www.gromacs.org/).
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:13]: - GROMACS: a simulation platform

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 3]
```diff
@@ -6,15 +6,24 @@
...
+- VMD:  a molecular visualization platform (https://www.ks.uiuc.edu/Research/vmd/).
+- Xvnc: a remote visualization tool to display the VMD platform remotely (https://en.wikipedia.org/wiki/X11vnc).
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:15]: tool -> library or program

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 4]
```diff
@@ -6,15 +6,24 @@
...
+The workflow requires:
+- Docker engine to be installed in the machines where GROMACS and VMD are executed.
+- Xvnc to be natively installed in the machine where Xvnc is executed.
+The workflow takes as input:
+- DISPLAY: display number used by Xvnc.
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:20]: what is display number ? we should clarify more this number I believe

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 5]
```diff
@@ -6,15 +6,24 @@
...
+- GROMACS_PORT: port number used by GROMACS to   stream the molecular simulation to VMD.
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:21]: there is extra space after `to`

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 2 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:13]: I've put the sentence that I took from Gromacs documentation

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 6]
```diff
@@ -6,15 +6,24 @@
...
+- GROMACS_MPI_THREADS: number of MPI threads to be executed by GROMACS.
+- GROMACS_OPENMP_THREADS: number of OPENMP threads per MPI thread of GROMACS. ]]>
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:23]: OpenMP

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 3 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:15]: I prefer a tool

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 6 above)
Comment (mboussaa) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:23]: per GROMACS MPI thread ?

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 4 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:20]: this is specific to xvnc, we can not explain xvnc in the workflow description

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 7]
```diff
@@ -26,10 +35,7 @@ VISU_COMMAND: vnc command to start ]]>
     
     fork="true">
       <description>
-        <![CDATA[ Example of Remote Visualization using Xvnc.
-It uses the following parameters:
-DISPLAY: display number to use
-VISU_COMMAND: vnc command to start ]]>
+        <![CDATA[ Start remote visualization using Xvnc. ]]>
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:38]: Start**s** as Stops ?

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 6 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml]: Done

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 7 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml]: Done

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 5 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml]: Done

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 2 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:13]: Done

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 1 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml]: Done

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml [hunk 8]
```diff
@@ -292,7 +292,7 @@ exit
     
     fork="true">
       <description>
-        <![CDATA[ A template of a signal task that waits until the value of the variable SIGNAL is added to the set of job signals (stored in ProActive signals channel). ]]>
+        <![CDATA[ A task that declares being ready for receiving the signal "Terminate_Job_and_Services", then iterates to check whether this signal is received (for example when sent by user from the Workflow Execution portal). When this signal is received, the task terminates. ]]>
```
Comment (mboussaa) [inline HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml:295]: - same as my remark of yesterday about iterates to ... => loop over to/until
- for example when sent by user from the Workflow Execution portal => is not really needed, and we are also going to include signals in Scheduling Portal

**Review by mboussaa — APPROVED** [review]

File: HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml (hunk 8 above)
Comment (nebilbenmabrouk) [reply HPC/resources/catalog/Gromacs_VMD_Molecule_Simulation.xml]: Done

**Pull request merged**